pkg/kubefed2/federate/util.go

```diff
@@ -49,3 +59,134 @@ func SetBasicMetaFields(resource *unstructured.Unstructured, apiResource metav1.
 		resource.SetNamespace(namespace)
 	}
 }
+
+func namespacedAPIResourceMap(config *rest.Config, skipAPIResourceNames string) (map[string]metav1.APIResource, error) {
+	apiResourceLists, err := enable.GetServerPreferredResources(config)
+	if err != nil {
+		return nil, err
+	}
+
+	apiResources := make(map[string]metav1.APIResource)
+	for _, apiResourceList := range apiResourceLists {
+		if len(apiResourceList.APIResources) == 0 {
+			continue
+		}
+
+		gv, err := schema.ParseGroupVersion(apiResourceList.GroupVersion)
+		if err != nil {
+			return nil, errors.Wrap(err, "Error parsing GroupVersion")
+		}
+
+		group := gv.Group
+		for _, apiResource := range apiResourceList.APIResources {
+			if !apiResource.Namespaced || isFederatedAPIResource(apiResource.Kind, group) ||
+				apiResourceMatchesSkipNames(apiResource, skipAPIResourceNames, group) {
+				continue
+			}
+
+			if previousAPIResource, ok := apiResources[apiResource.Kind]; ok {
+				compared, err := compareGroupVersions(gv, schema.GroupVersion{Group: previousAPIResource.Group, Version: previousAPIResource.Version}, apiResource.Kind)
+				if err != nil {
+					return nil, err
+				}
+				if compared <= 0 {
+					// The newer GV is not latest keep the previous.
+					continue
+				}
+			}
+
+			// The individual apiResources do not have the group and version set
+			apiResource.Group = group
+			apiResource.Version = gv.Version
+
+			apiResources[apiResource.Kind] = apiResource
+		}
+	}
+
+	return apiResources, nil
+}
+
+func compareGroupVersions(newGV, oldGV schema.GroupVersion, kind string) (int, error) {
+	// The strategy involved to choose a Group higher in order here is to
+	// consider 'extensions' as the outdated group [This seems to be true for
+	// all k8s APIResources, so far]. For example if deployments exists in
+	// 'extensions' and 'apps'; 'deployments.apps' will be chosen.
+	// This doesn't apply to events but events are listed in controllerCreatedAPIResourceNames
+	// and skipped always.
+	if oldGV.Group != newGV.Group && (oldGV.Group == "extensions" || newGV.Group == "extensions") {
```

Comment:
I think it would be preferable to simply ignore the `extensions` group in `namespacedAPIResourceMap`:

```bash
group = gv.Group
// The `apps` group has been superseded by the `extensions` group 
if group == "extensions" {
    continue
}
```

Reply:
This wont work in the scenario if there is only one group exposed on an API server and that be `extensions`. Some API installation might choose to do that, isn't it?

Reply:
`extensions` is deprecated.  We are under no obligation to support deprecated APIs.